### PR TITLE
feat: Five Most Popular Courses Chart

### PR DESCRIPTION
**Pull request by piotr-pajak**
## Jira issue(s)
[LC-410](https://selleolabs.atlassian.net/browse/LC-410)

## Overview  
- Implemented the `FiveMostPopularCoursesChart` component.  
- Developed the `useMediaQuery` hook with comprehensive unit test coverage.  
- Added the `useAdminStats` hook to handle data fetching.  

## Screenshots
Mobile:
Screenshot 2024-12-02 at 07 56 19

Desktop:
Screenshot 2024-12-02 at 07 56 02

[LC-410]: https://selleolabs.atlassian.net/browse/LC-410?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/web/app/api/queries/useAdminStatistics.ts
Comment (wielopolski) [inline apps/web/app/api/queries/useAdminStatistics.ts:1]: Its teacher statistics, not `user` or `admin`

File: apps/web/app/modules/Dashboard/DashboardNavigation/DashboardNavigation.tsx
Comment (wielopolski) [inline apps/web/app/modules/Dashboard/DashboardNavigation/DashboardNavigation.tsx:1]: what is this file?

File: apps/web/app/modules/Statistics/Admin/components/FiveMostPopularCoursesChart.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                  x={0}
+                  textAnchor="middle"
+                  verticalAnchor="middle"
+                  className="fill-primary-950 h5 md:h3 translate-x-1/2 translate-y-1/2"
+                >
+                  No data available
+                </Text>
+              ) : null;
+            }}
+          />
```
Comment (typeWolffo) [inline apps/web/app/modules/Statistics/Admin/components/FiveMostPopularCoursesChart.tsx:89]:
```suggestion
          <Customized
            component={() => 
              isEmptyChart && (
                <Text
                  x={0}
                  textAnchor="middle"
                  verticalAnchor="middle"
                  className="fill-primary-950 h5 md:h3 translate-x-1/2 translate-y-1/2"
                >
                  No data available
                </Text>
              );
            }
          />
```

what do you think about that?

File: apps/web/app/types/shared.ts [hunk 2]
```diff
@@ -6,3 +6,12 @@ export type SVGComponentProps = SVGProps<SVGSVGElement>;
...
 export type CourseListLayout = "table" | "card";
+
+export type AtLeastOne<T> = {
+  [K in keyof T]-?: Pick<T, K> & Partial<Omit<T, K>>;
+}[keyof T];
+
+export type ToMediaQueryObjectParam = AtLeastOne<{
+  minWidth?: number;
+  maxWidth?: number;
+}>;
```
Comment (typeWolffo) [inline apps/web/app/types/shared.ts:17]: nice, but probably we can keep it simple with discriminated union
```ts
export type ToMediaQueryObjectParam = {
  minWidth: number;
  maxWidth?: number;
} | {
  minWidth?: number;
  maxWidth: number;
};
```

**Review by typeWolffo — CHANGES_REQUESTED** [review]
🟢 but I left some comments

**Pull request merged**